tests/zenko_tests/node_tests/iam_policies/cloudserver/AssumeRoleWithWebIdentity.js

```diff
@@ -0,0 +1,144 @@
+
+const clientAdmin = VaultClient.getAdminClient();
+const accountName = 'AccountTest';
+const accountInfo = {
+    email: `${accountName}@test.com`,
+    password: 'test',
+};
+const externalAccessKey = 'DZMMJUPWIUK8IWXRP0HQ';
+const externalSecretKey = 'iTuJdlidzrLipymvAGrLP66Yxghl4NQxLZR3cLlu';
+
+const duration = '1000';
+
+const bucket1 = "bucket1";
+const bucket2 = "bucket2";
+
+const storageManagerName = 'storage_manager';
+const storageAccountOwnerName = 'storage_account_owner';
+const dataConsumerName = 'data_consumer';
+const storageManagerRole = 'storage-manager-role';
+const storageAccountOwnerRole = 'storage-account-owner-role';
+const dataConsumerRole = 'data-consumer-role';
+
+describe('iam policies - cloudserver - AssumeRoleWithWebIdentity - Metadata', () => {
+
+    before(done => {
+        async.series([
+            // create an account, generateAccountAccessKey for it, get iam client, sts client and s3 client of this account
+            next => clientAdmin.createAccount(accountName, accountInfo, next),
+            next => clientAdmin.generateAccountAccessKey(accountName, next, { externalAccessKey, externalSecretKey }),
+            next => {
+                iamClient = VaultClient.getIamClient(externalAccessKey, externalSecretKey);
+                stsClient = getSTSClient(externalAccessKey, externalSecretKey);
+                s3Client = getS3Client(externalAccessKey, externalSecretKey);
+                next();
+            },
+            //use s3 client to create 2 buckets and put 2 objects respectively
+            next => {
+                async.series([
+                    next => s3Client.createBucket({Bucket: bucket1}, next),
+                    next => s3Client.putObject({Bucket: bucket1, Key: "file1"}, next),
+                    next => s3Client.putObject({Bucket: bucket1, Key: "file2"}, next),
+                    next => s3Client.createBucket({Bucket: bucket2}, next),
+                    next => s3Client.putObject({Bucket: bucket2, Key: "file1"}, next),
+                    next => s3Client.putObject({Bucket: bucket2, Key: "file2"}, next),
+                ], next)
+            },
+        ], done)
+    });
+
+    after(done => {
+        async.series([
+            next => s3Client.deleteObjects({
+                Bucket: bucket1,
+                Delete: {
+                    Objects: [{Key: 'file1'}, {Key: 'file2'}],
+                    Quiet: false,
+                },
+            }, next),
+            next => s3Client.deleteObjects({
+                Bucket: bucket2,
+                Delete: {
+                    Objects: [{Key: 'file1'}, {Key: 'file2'}],
+                    Quiet: false,
+                },
+            }, next),
+            next => s3Client.deleteBucket({Bucket: bucket1}, next),
+            next => s3Client.deleteBucket({Bucket: bucket2}, next),
+            next => clientAdmin.deleteAccount(accountName, next),
+        ], done)
+    });
+
+
+    tests = [
```

Comment:
I feel like you are only testing seeded roles, maybe you can test custom roles as well ? 
And when you explicitly deny the operation, for example, you are allowed to mdsearch on all bucket but you have a deny on `bucket1`
And if you only have the right, let's say to listbucket, will you get a 403 when you mdsearch ?

Reply:
for testing custom roles, we need to change the default seting in [keycloak configuration](https://github.com/scality/Zenko/blob/development/2.3/eve/workers/end2end/scripts/configs/keycloak_config.json) . by default keycloak only has `AccountTest::DataConsumer` group, `AccountTest::StorageAccountOwner`group and `StorageManager` role and 3 corresponding users: `data_consumer`, `storage_account_owner` and `storage_manager`. So I can only test using these 3 keycloak users